README.md

```diff
@@ -1,55 +1,185 @@
+this definition in one place that can be found and updated easily in the future.
+Win, win, win. Are we beginning to see the power off presentational components? 
+Good.
 
-It's easy to go from here to building a `<Header />`, `<Footer />` etc.
+Now imagine that it's not just the `TextField` that our team has executed
+in this way, but say we've also defined a `<Header />` and a `<Footer />`, as
+well, of course, as other more unique and customized modules that are still
+primarily presentational. Imagine further that we've composed the majority of our
+UI out of these simple presentational components -- all of them almost entirely
+stateless, all of them designed to do one thing and one thing well: they just
+receive `props` from their parent components and render! That's it. They are
+simple and beautiful and because they aren't doing much, because they are mostly
+stateless, they have a better chance of remaining blissfully bug free!
 
-Finally, we take away all of the crusty `class` boilerplate and show students
-that we can just write, e.g.,
+This is the power and importance of presentational components. They are simple
+and they just work. So therefore we should strive to use them as much as possible.
+
+## The "Stateless Functional" Component & "Pure" Functions
+
+What if I told you we can actually make our presentational components even
+simpler? Well we can, and here's why: Remember how one of the principles in
+our checklist for the presentational component pattern was that the component
+(probably) does not have state? Well, if in fact we can create a component
+that has no state, that means that our component doesn't even really need to be
+a Javascript object of type `Component` at all. It can just be a simple function
+&mdash; -- one that takes an input and returns a (portion of) the UI.
+
+So what's this look like? Here's our `TextField` component rendered as a so-called
+`functional stateless` component (a feature available in React since v0.14):
+
+```
+const defaultLimit = 100
+const noop = () => {}
 
-```javascript
 export default function TextField(props) {
-    return <input className="field field-light"
-        onChange={this.props.onChange || noop}
-        limit={this.props.limit || defaultLimit} />
+	return <input
+					 className="field field-light"
+					 onChange={this.props.onChange || noop}
+					 limit={this.props.limit || defaultLimit} />
 }
 ```
+Now isn't that just beautiful? It really is. It's just so concise. We've discarded
+all that ugly boilerplate. But it's not only concision that makes this beautiful. 
+By transforming our component into a stateless function, we have made our
+`TextField` component an extremely stable and predictable part of our application.
 
-## Resources
+The predictability comes from the fact -- and here we can see the influence on
+React of the principles of functional programming -- that this function will
```

Comment:
'and here we can see the influence on React of the principles of functional programming' -> 'and here we can see the influence of functional programming on React'? Just a suggestion, feels like this improves the readability